File: frontend/src/api/community.ts
import api from './client';

export interface CommunityPostUser {
  id: string;
  name: string | null;
  display_name: string | null;
}

export interface CommunityPost {
  id: string;
  user_id: string;
  content: string;
  is_spam: boolean;
  created_at: string;
  updated_at: string;
  user: CommunityPostUser;
}

export interface CommunityPostList {
  total: number;
  posts: CommunityPost[];
}

export const communityApi = {
  getPosts: (skip = 0, limit = 20) =>
    api.get<CommunityPostList>('/api/v1/community/', { params: { skip, limit } }),
  createPost: (content: string) =>
    api.post<CommunityPost>('/api/v1/community/', { content }),
  deletePost: (postId: string) =>
    api.delete(`/api/v1/community/${postId}`),
};

File: backend/app/schemas/community.py
from datetime import datetime
from pydantic import BaseModel, Field, ConfigDict


class CommunityPostUser(BaseModel):
    id: str
    name: str | None = None
    display_name: str | None = None


class CommunityPostCreate(BaseModel):
    content: str = Field(..., min_length=1, max_length=1000)


class CommunityPostOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    user_id: str
    content: str
    is_spam: bool
    created_at: datetime
    updated_at: datetime
    user: CommunityPostUser


class CommunityPostList(BaseModel):
    total: int
    posts: list[CommunityPostOut]
